assets/js/main.js

```diff
@@ -156,13 +150,37 @@ $(document).ready((function ($) {
         return false;
     });
 
+    $('.appointment-button').click(function () {
+        var $form = $('.appointment-form'),
+            data = getFormData($form),
+            host = 'http://127.0.0.1:8000/',
+            path = 'api/appointment/',
+            url = host + path;
+        $.ajax({
+            type: 'POST',
+            url: url,
+            data: data,
+            beforeSend: removeFieldErrors,
+            success: function () {
+                $('.alert-success').fadeIn(function () {
+                    $(this).fadeOut(14000);
+                });
+                $form.find('input').each(function () {
+                    $(this).val('');
+                    $('textarea').val('');
+                    $('select').val('');
+                });
+            },
+            error: showFieldErrors,
+        });
+        return false;
+    });
+
```

Comment:
Now when you submit an empty form an exception happens in the backend and the frontend has no reaction to that. 

You need to handle an empty appointment form submit it in a way that backend doesn't throw 500.

Essentially it all comes down to properly handling a situation when you receive an empty `date` variable. 

<img width="1283" alt="Screenshot 2021-12-31 at 00 11 58" src="https://user-images.githubusercontent.com/12160621/147791268-c375972c-840d-45a3-a463-868cbe0485ae.png">

Reply:
Done